sqlalchemy-stubs/sql/functions.pyi

```diff
@@ -44,7 +48,7 @@ class FunctionElement(Executable, ColumnElement[_T], FromClause, Generic[_T]):
     def select(self) -> Select: ...  # type: ignore  # incompatible with FromClause.select
     def scalar(self) -> Any: ...  # type: ignore  # incompatible with Executable.scalar
     def execute(self) -> ResultProxy: ...  # type: ignore  # incompatible with Executable.execute
-    def self_group(self, against: Optional[Any] = ...) -> Union[ColumnElement[_T], Grouping[_T]]: ...
+    def self_group(self: _FE, against: Optional[Any] = ...) -> Union[AsBoolean, Grouping[_T], _FE]: ...
```

Comment:
Same comment about union return here. If people will complain, then we can replace this with `Any`.